test/test_sparse_semi_structured.py

```diff
@@ -233,6 +233,21 @@ def test_cslt_sparse_mm_int8_in_fp16_out(self, device):
             sparse_result = torch._cslt_sparse_mm(A_sparse.compressed_tensor_cusparselt, B.t(), out_dtype=torch.float16)
             assert torch.allclose(dense_result, sparse_result, rtol=1e-3, atol=1e-3)
 
+    def test_cslt_sparse_mm_int8_in_int32_out(self, device):
+        """
+        Test sparse mam with int8 input with int32 output for cuSPARSELt
+        """
+        if "cusparselt" in SEMI_STRUCTURED_SUPPORTED_BACKENDS:
+            SparseSemiStructuredTensor._FORCE_CUTLASS = False
+            A = rand_sparse_semi_structured_mask(128, 128, dtype=torch.int8)
+            A_sparse = to_sparse_semi_structured(A)
+
+            B = torch.rand((128, 128), device=A_sparse.device).to(torch.int8)
+
+            dense_result = torch.mm(A.cpu(), B.t().cpu()).to(device, dtype=torch.int64)
```

Comment:
I still don't think this fixes it.

You'd have to do 
```
dense_result = torch.mm(A.to(torch.int64).cpu(), B.to(torch.int64).cpu()).to(device)
assert dense_result.dtype = torch.int64
```

Reply:
I don't see how else we can check that indeed cuSPARSElt is now returning i32 instead of i8